app/js/lib/foxbox/settings.js

```diff
@@ -22,178 +21,203 @@ const API_VERSION = 1;
+    key: 'watchInterval',
+    type: 'number',
+    defaultValue: 3000,
+  }),
+  ONLINE_CHECKING_INTERVAL: Object.freeze({
+    key: 'onlineCheckingInterval',
+    type: 'number',
+    defaultValue: 5000,
+  }),
+  ONLINE_CHECKING_LONG_INTERVAL: Object.freeze({
+    key: 'onlineCheckingLongInterval',
+    type: 'number',
+    defaultValue: 1000 * 60 * 5,
+  }),
+
+  // String settings.
+  LOCAL_ORIGIN: Object.freeze({ key: 'localOrigin' }),
+  TUNNEL_ORIGIN: Object.freeze({ key: 'tunnelOrigin' }),
+  CLIENT: Object.freeze({ key: 'client' }),
+  SESSION: Object.freeze({ key: 'session' }),
+  PUSH_ENDPOINT: Object.freeze({ key: 'pushEndpoint' }),
+  PUSH_PUB_KEY: Object.freeze({ key: 'pushPubKey' }),
+  REGISTRATION_SERVICE: Object.freeze({
+    key: 'registrationService',
+    defaultValue: 'https://knilxof.org:4443/ping',
+  }),
+});
 
 export default class Settings extends EventDispatcher {
-  constructor() {
-    super(['session', 'polling-setting']);
-
-    const localOrigin = storage.getItem(`${PREFIX}localOrigin`);
-
-    const pollingEnabled = storage.getItem(`${PREFIX}pollingEnabled`) !== null ?
-      storage.getItem(`${PREFIX}pollingEnabled`) === 'true' :
-      DEFAULT_POLLING_ENABLED;
-
-    this._configured = storage.getItem(`${PREFIX}configured`) !== null ?
-      storage.getItem(`${PREFIX}configured`) === 'true' : false;
-    this._localOrigin = localOrigin;
-    this._tunnelOrigin = storage.getItem(`${PREFIX}tunnelOrigin`) || '';
-    this._client = storage.getItem(`${PREFIX}client`) || '';
-    this._session = storage.getItem(`${PREFIX}session`);
-    this._skipDiscovery = storage.getItem(`${PREFIX}skipDiscovery`) === 'true';
-    this._pollingEnabled = pollingEnabled;
-    this._pushEndpoint = storage.getItem(`${PREFIX}push_endpoint`) || null;
-    this._pushPubKey = storage.getItem(`${PREFIX}push_pubKey`) || null;
-  }
+  constructor(storage = localStorage) {
+    super();
+
+    // Not all browsers have localStorage supported or activated.
+    this[p.storage] = storage || {
+      getItem: () => null,
+      setItem: () => {},
+      removeItem: () => {},
+      clear: () => {},
+    };
+
+    this[p.values] = new Map();
+
+    Object.keys(settings).forEach((settingName) => {
+      const setting = settings[settingName];
+      const settingStringValue = this[p.storage].getItem(
+        `${PREFIX}${setting.key}`
+      );
+
+      // Setting values directly to avoid firing events on startup.
+      this[p.values].set(
+        setting,
+        this[p.stringToSettingTypedValue](setting, settingStringValue)
+      );
+    });
 
-  clear() {
-    return new Promise((resolve) => {
-      // @todo Remove only the items set here.
-      storage.clear();
-      this._session = null;
+    window.addEventListener('storage', this[p.onStorage].bind(this));
 
-      resolve();
-    });
+    Object.seal(this);
   }
 
-  get configured() {
-    return this._configured;
+  get [settings.CONFIGURED.key]() {
```

Comment:
Yes, I like the benefit of having a clean doc generated automatically. Let's use static name for the getters and setters.